### PR TITLE
fix: Built a standardized way for the return of errors in 2Factor

### PR DESCRIPTION
**Pull request by ArneGudermann**

### CODE REVIEW

File: src/viur/core/modules/user.py [hunk 1]
```diff
@@ -773,14 +775,14 @@ def otp(self, *args, **kwargs):
         if res is None:
             otp_user_conf["attempts"] = attempts + 1
             session.markChanged()
-
+            skel = self.OtpSkel()
```
Comment (phorward) [inline src/viur/core/modules/user.py:778]: You can re-use `skel` defined in line 760 for this.
```suggestion
```

File: src/viur/core/modules/user.py [hunk 2]
```diff
@@ -773,14 +775,14 @@ def otp(self, *args, **kwargs):
...
+            skel.errors = [ReadFromClientError(ReadFromClientErrorSeverity.Invalid, "Wrong OTP Token", ["otptoken"])]
```
Comment (phorward) [inline src/viur/core/modules/user.py:779]: Error shall not be set on first call.
```suggestion
            if attempts > 0:
                skel.errors = [ReadFromClientError(ReadFromClientErrorSeverity.Invalid, "Wrong OTP Token", ["otptoken"])]
```

File: src/viur/core/modules/user.py [hunk 3]
```diff
@@ -1014,7 +1018,7 @@ def authenticator_otp(self, **kwargs):
         if AuthenticatorOTP.verify_otp(otp=otp_token, secret=user["otp_app_secret"]):
             return self._user_module.secondFactorSucceeded(self, user_key)
 
-        skel.errors = [ReadFromClientError(ReadFromClientErrorSeverity.Invalid, "Wrong OTP Token")]
+        skel.errors = [ReadFromClientError(ReadFromClientErrorSeverity.Invalid, "Wrong OTP Token", ["otptoken"])]
```
Comment (phorward) [inline src/viur/core/modules/user.py:1021]: The handler `AuthenticatorOTP` doesn't seem to hold a counter for a maximum retry call. Can you please fix this, as it's the case in `TimeBasedOTP`.

**Review by phorward — CHANGES_REQUESTED** [review]

File: src/viur/core/modules/user.py (hunk 2 above)
Comment (ArneGudermann) [reply src/viur/core/modules/user.py:779]: But why ? We start with the `start` method and send the token and the we must send an error back ?

Comment (phorward) [reply src/viur/core/modules/user.py:777]: OK it seems you're right.

**Review by phorward — APPROVED** [review]

**Pull request merged**